air/src/farewell_step/outcome.rs

```diff
@@ -104,12 +112,23 @@ fn populate_outcome_from_contexts(
         "serde_json::to_vec(call_results)",
     );
 
-    InterpreterOutcome {
-        ret_code,
-        error_message,
-        data,
-        next_peer_pks,
-        call_requests,
+    InterpreterOutcome::new(ret_code, error_message, data, next_peer_pks, call_requests)
+}
+
+// this method is called only if there is an internal error in the interpreter and
+// new execution trace was corrupted
+fn execution_error_into_outcome(
```

Comment:
This comment is wrong. This method is called always and it decides if it was an error or not. I suggest something like this in the calling code:
```match maybe_gens.map_err(execution_error_into_outcome)```
Then this comment will decribe reality properly.